Generalize onboarding process (#1289)
sourcegraph/about: handbook/engineering/onboarding.md

# Onboarding

Welcome! We're excited to have you join the team. This document outlines the structure of your first few weeks at Sourcegraph.

## Guiding principles

### There are no stupid questions

Joining a new company can be overwhelming — there's a lot to learn! As you navigate your first few weeks at Sourcegraph, we want you to know that everyone on the team is here to help, and that there are **no stupid questions**.

Every time you're curious or confused about something — just ask! When you do so, use [public discussion channels](../communication/team_chat.md#avoid_private_messages) as much as possible.

### Think and act like an owner

> At Sourcegraph, we don't think of engineers as resources — we think of them as owners of their work, who constantly reevaluate how to use their talents to be as impactful as possible. We value your opinions and ideas. You should always feel empowered to identify potential improvements and act upon them, whether they be improvements to this onboarding process, our handbook and general documentation, our codebase and tooling, or our product.

Reviewer: ```suggestion
At Sourcegraph, we don't think of engineers as resources — we think of them as owners of their work, who constantly reevaluate how to use their talents to be as impactful as possible. We value your opinions and ideas. You should always feel empowered to identify potential improvements and act upon them, whether they be improvements to processes (like onboarding), our handbook and general documentation, our codebase and tooling, or our product.
```

Never assume that a problem is somebody else's to fix!

## Getting set up

You'll have to get some basics set up in your first few days:

> - Complete [general onboarding](../people-ops/from-graphbook/onboarding.md#for-all-new-teammates)

Author: ```suggestion
- Complete [general onboarding](../people-ops/onboarding/index.md#for-all-new-teammates)
```

- [Configure your GitHub notifications.](./github-notifications/index.md)

> - Join #dev-announce, #dev-chat, and your team's channel on Slack. [Team chat documentation](../communication/team_chat.md#engineering)

Reviewer: ```suggestion
- Join #dev-announce, #dev-chat, and your team's channel on Slack, as well as any other channels you find interesting. [Team chat documentation](../communication/team_chat.md#engineering)
```

  - [Site24x7](https://www.site24x7.com) (optional; recommended for Distribution team members)
  - [HoneyComb.io](https://www.honeycomb.io/)
  - Ask Christina to send an invite to [Productboard](https://sourcegraph.productboard.com)

## Weeks 1-3

### Starter tasks

Your manager will assign to you three starter tasks that you should aim to complete in your first three weeks. These tasks are small, atomic, and intended to expose you to different parts of our codebase and product: it's important that you build the flexibility to fix any problem you'll be faced with at Sourcegraph, and don't narrow down your comfort zone to a certain part of our codebase or product.

> As you're working on these tasks:

Reviewer 2: When I joined @beyang said to me: "In your first few weeks: optimise for learning" @nicksnyder said something along the same lines and it really stuck with me. It takes a lot of the "i need to ship asap" pressure of when starting.

Maybe we can add something like this to this section?

Author: What do you think of this?

```suggestion
As you're working on these tasks:
- Optimize for learning: your priority is to soak up as much context as possible.
```

Reviewer 2: Yep, that's good, or go even further:

```suggestion
As you're working on these tasks:
- Optimize for learning: your priority is to soak up as much context as possible, not to ship something as fast as possible.
```

I'll let you decide, though.

- Chat with the teammates currently working on that goal, and relevant stakeholders (product, design, CE) to understand the problem being solved.
- Prepare a proposal for how you'll work with the team solving that problem in the following three weeks. Your work may include:
    - Working on previously identified GitHub issues
    - Doing spikes to solve unknowns and learn more about the problem
    - Writing RFCs to propose solutions, and planning resulting development work
- Discuss your plan with the team and your manager.
- Get hacking!

At Sourcegraph, you'll be expected to own the problems your team is solving, and work with the team to define solutions and plan your work — we think it's important that you start doing this early on!

### Give feedback on your onboarding

At the end of week 6 at Sourcegraph, your manager will send you a survey to learn more about what worked and what didn't during your onboarding. Take your time to answer this thoughtfully — your answers will be very important to make sure our onboarding process is even better for future hires!

> TODO @lguychard: prepare this

Reviewer: ```suggestion
```
